### ISSUE TITLE
Arch bumblebee package

### ISSUE THREAD

**Issue by AntouanK** [tobi-wan-kenobi/bumblebee-status#172]
Not sure if anyone can help on that, but on arch, the bumblebee-status package is on AUR as [`bumblebee-status-git`](https://aur.archlinux.org/packages/bumblebee-status-git/).
The annoying thing is that it always asks to update, because it has to check the repo or something.
Can anyone make a "normal" package for it? So it updates like the others?

Not sure how to do that and if the author of the repo should do it (to be "official").

**Comment by tobi-wan-kenobi** [comment]
Have to look into that. Unfortunately, I'm not using Arch myself, so it might take some time to have a setup and wrap my head around it :)

**Comment by yvesh** [comment]
Someone should do a general AUR package, which is not using git master, but releases..

I can help with creating the initial AUR project and creating build files etc.. But I probably won't have time to maintain it on every update.

**Comment by AntouanK** [comment]
@yvesh Is there a way to automate it? So when @tobi-wan-kenobi makes a release in the github repo, to trigger something to update the arch package as well.

**Comment by tobi-wan-kenobi** [comment]
@yvesh If I know how to create the AUR, I'll gladly take over the creation whenever I do a new release. I think that makes most sense, because I "trigger" the release anyhow.

@AntouanK Once this is up and running, maybe it pays off to look into automation via Travis. I think it *should* be possible to pick up new releases there.... Hopefully :P

**Comment by AntouanK** [comment]
@tobi-wan-kenobi Travis is free for open-source projects, so yeah, make use of it. :)

If you need anyone to test the AUR package, ping me.

**Comment by yvesh** [comment]
@AntouanK you always have to update the hash sums (like sha256 / md5 etc.) in the PKGBUILD and check if the build scripts need adjustments (Shouldn't be much probably). 

You can point to the latest release package here on GitHub in the PKGBUILD, the downside is if the hashes are not updated fast  after a release you screw the package with that and have an ugly error. So either update the package fast or have it linked to release versions. (Than you also have to update the URL for each release).

@tobi-wan-kenobi okay, i try getting that done in the next days :) 

Not sure if how much work it is with Travis, i know some people doing it with a Jenkins deploy job.. You need to push to a repo for updating the PKGBUILD and do some replacements in the PKGBUILD at least. But that's actually something that should be possible with Travis too. (Personally never used Travis for deployment, but only for running tests for every PR / commit).

**Comment by tobi-wan-kenobi** [comment]
That sounds great, thanks a lot to everyone involved!

bumblebee-status already uses Travis for unit tests, so I think it *should* be possible to trigger a job on a new release.

**Comment by tobi-wan-kenobi** [comment]
Hi,

looking at https://github.com/glaxx/aur-bumblebee-status-git/blob/master/PKGBUILD, I *think* I was able to come up with a PKGBUILD that always builds the latest release version. However, since I don't have an Arch Linux, dependencies are hell for me....

Could someone maybe have a look at it and the experimental package I built? Does this make sense?

**Comment by yvesh** [comment]
Hi @tobi-wan-kenobi,
great that you looked at it, imo the python dependency list from the other package is fine.. Probably python-dbus and python-dbus-common are missing. IIRC some packages weren't available in arch and I had to install them with pip (not the default modules iirc).

The python libs are all running without X, so maybe adding X wouldn't be a bad idea.. But i don't think you really need to care that much on dependencies. You can always update the PKGBUILD anytime and arch users are pretty fast at reporting issues ;-)

Just did a test build and installation, both went fine and it's running.

Thank you!

**Comment by tobi-wan-kenobi** [comment]
Short update: Since I know next to nothing about Arch myself, I've sent the PKGBUILD to the arch mailing list for double-checking. As soon as that's cleared, I'll upload it to the AUR servers.

**Comment by AntouanK** [comment]
@tobi-wan-kenobi I just managed to get `bumblebee-status` installed.
For some reason `currency` module is not working.
I get this 
```
Traceback (most recent call last):
  File "/usr/share/bumblebee-status/bumblebee/engine.py", line 168, in _load_module
    module = importlib.import_module("bumblebee.modules.{}".format(module_name))
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 978, in _gcd_import
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load
  File "<frozen importlib._bootstrap>", line 948, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'bumblebee.modules.currency'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/bumblebee-status", line 66, in <module>
    main()
  File "/usr/bin/bumblebee-status", line 38, in main
    inp=inp,
  File "/usr/share/bumblebee-status/bumblebee/engine.py", line 118, in __init__
    self.load_modules(config.modules())
  File "/usr/share/bumblebee-status/bumblebee/engine.py", line 134, in load_modules
    mod = self._load_module(module["module"], module["name"])
  File "/usr/share/bumblebee-status/bumblebee/engine.py", line 170, in _load_module
    raise bumblebee.error.ModuleLoadError(error)
bumblebee.error.ModuleLoadError: No module named 'bumblebee.modules.currency'
```

If I omit `currency`, it seems to be working fine.
Is that expected?

**Comment by tobi-wan-kenobi** [comment]
Ah, yes, that's expected :-P

The "problem" is: The AUR package in question is built from the latest released version of bumblebee-status (1.4.2 at the moment), which did not include the `currency` module yet.

If you want to have that, you'll have to use the `bumblebee-status-git` package provided by @glaxx, or wait until I get around to releasing a new version (unfortunately, `master` seems to have a few open issues at the moment).

Hope that answers your question!

**Comment by AntouanK** [comment]
@tobi-wan-kenobi Got it.
I can live for a few days/weeks without currency then :)
